R/dictionary_based_cleaning.R

```diff
@@ -25,30 +39,55 @@ clean_using_dictionary <- function(data, dictionary) {
                                null.ok = FALSE)
   checkmate::assert_data_frame(dictionary, min.rows = 1L, max.cols = 4L,
                                null.ok = FALSE)
-  stopifnot(all(c("options", "values", "grp") %in% names(dictionary)),
-            all(unique(dictionary[["grp"]]) %in% names(data)))
+  # abort if the provided data dictionary does not contain the following three
+  # column names: options, values, grp.
+  if (!all(c("options", "values", "grp") %in% names(dictionary))) {
+    all_columns <- c("options", "values", "grp", "orders") # nolint: object_usage_linter
+    mandatory_columns <- c("options", "values", "grp") # nolint: object_usage_linter
```

Comment:
```suggestion
    mandatory_columns <- c("options", "values", "grp")
    all_columns <- c(mandatory_columns, "orders") # nolint: object_usage_linter
```

To reduce duplication

Reply:
Good point - thanks